dd-java-agent/agent-profiling/profiling-uploader/src/main/java/com/datadog/profiling/uploader/RecordingUploader.java

```diff
@@ -347,4 +338,57 @@ private boolean canEnqueueMoreRequests() {
         .map(e -> e.getKey() + ":" + e.getValue())
         .collect(Collectors.toList());
   }
+
+  private class RetryCallback implements Callback {
+    private int nbRetries = 0;
+
+    @SneakyThrows(IOException.class)
+    private boolean tryRetryRequest(Call call) {
+      // Don't retry if the queue is full to accept new recordings
+      if (canEnqueueMoreRequests() && nbRetries++ < MAX_RETRIES) {
+        // Retry after 1s, 2s, 4s, 8s
+        long backoffSec = 1 << nbRetries;
+        if (backoffSec > 8)
+          backoffSec = 8;
+
+        try {
+          Thread.sleep(backoffSec * 1_000L);
+        } catch (InterruptedException e) {
+          // No special action here, we still have to send the request anyway
+        }
+        client.newCall(call.request()).enqueue(this);
+        return true;
+      }
+      currentInflightBytes.addAndGet((int) (-1 * call.request().body().contentLength()));
+      return false;
+    }
+
+    @Override
+    public void onFailure(final Call call, final IOException e) {
+      if (tryRetryRequest(call)) {
+        log.error("Failed to upload recording, retry upload" + nbRetries + " time(s)", e);
```

Comment:
Please, use the logger formatting placeholders 
```suggestion
        log.error("Failed to upload recording, retry upload {} time(s)", nbRetries, e);
```